Fix #3634 Fix issues with Block.rotateBlock

mezz opened the pull request:
See #3634 for more information.

mezz committed:
Fix #3634 Stop rotation of extended pistons, beds, and double chests

LexManos commented on patches/minecraft/net/minecraft/block/BlockBed.java.patch:
```diff
@@ -0,0 +1,13 @@
+     }
++
++    /* ======================================== FORGE START =====================================*/
++    public boolean rotateBlock(World world, BlockPos pos, EnumFacing axis)
```
Rather then expand the number of patches we have, it's preferrible to add this type of stuff to the base Block class.

liach commented on src/test/java/net/minecraftforge/debug/WrenchRotateDebug.java:
```diff
@@ -0,0 +1,106 @@
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+
+@Mod(modid = WrenchRotateDebug.modID, name = "Wrench Rotate Debug", version = "0.0.0", acceptableRemoteVersions = "*")
```
`acceptableRemoteVersions = "*"` seems wrong.

mezz committed:
Remove acceptableRemoteVersions

mezz committed:
Remove patches for BlockBed and BlockPistonExtension

mezz committed:
Fix rotateBlock to cycle in the correct order

mezz committed:
Fix rotation of repeaters and comparators

mezz commented:
Added fixes for rotating comparators and repeaters, they were not updating their state when rotated before.

Added fixes for rotation order. Before it was pretty random (cycled the property).
Now the behavior is:
 * If block has only horizontal facings, rotate it clockwise
   * much more intuitive than cycling North, South, West, East
 * If block can be rotated on the face it was clicked, do so 
   * this is what the javadoc says the method does
 * Otherwise (block has weird allowed facings) just cycle the property
    * better than nothing, this is the current 1.11 behavior
